poutyne/framework/callbacks/mlflow_logger.py

```diff
@@ -0,0 +1,241 @@
+                             seed=42, callbacks=[mlflow_logger])
+    """
+
+    def __init__(self,
+                 experiment_name: str,
+                 tracking_uri: Union[str, None] = None,
+                 batch_granularity: bool = False) -> None:
+        super().__init__(batch_granularity=batch_granularity)
+        if mlflow is None:
+            raise ImportError("Mlflow needs to be installed to use this callback.")
+
+        self.tracking = tracking_uri
+
+        self._working_directory = os.getcwd()  # for Git
+
+        self.ml_flow_client = MlflowClient(tracking_uri=self.tracking)
+
+        self._handle_experiment_id(experiment_name)
+        self.run_id = self.ml_flow_client.create_run(experiment_id=self.experiment_id).info.run_id
+
+        self._log_git_version()
+
+        self._status = "FAILED"  # base case is a failure
+
+    def log_config_params(self, config_params: Union[Mapping, Sequence]) -> None:
+        """
+        Args:
+            config_params (Union[Mapping, Sequence]):
+                The config parameters of the training to log, such as number of epoch, loss function, optimizer etc.
+        """
+        if isinstance(config_params, Dict):
+            mlflow.log_params(config_params)
+        else:
+            for param_name, element in config_params.items():
+                self._log_config_write(param_name, element)
+
+    def log_param(self, param_name: str, value: Union[str, float]) -> None:
+        """
+        Log the value of a the parameter into the last run of the experiment.
+
+        Args:
+            param_name (str): The name of the parameter.
+            value (Union[str, float]: The value of the parameter.
+
+        """
+        self.ml_flow_client.log_param(run_id=self.run_id, key=param_name, value=value)
+
+    def log_metric(self, metric_name: str, value: float, step: Union[int, None] = None) -> None:
+        """
+        Log the value of a metric into the last run of the experiment.
+
+        Args:
+            metric_name (str): The name of the metric.
+            value (float): The value of the metric.
+            step (Union[int, None]): The step when the metric was compute (Default = None).
+        """
+        self.ml_flow_client.log_metric(run_id=self.run_id, key=metric_name, value=value, step=step)
+
+    def _log_config_write(self, parent_name: str, element: Union[int, float, str, Mapping, Sequence]) -> None:
+        """
+        Log the config parameters when it's a mapping or a sequence of elements.
+        """
+        if isinstance(element, Mapping):
+            for key, value in element.items():
+                # We recursively open the element (Dict format type)
+                self._log_config_write("{}.{}".format(parent_name, key), value)
+        elif isinstance(element, Sequence) and not str:  # Since string are sequence we negate it to be log in the else
```

Comment:
```suggestion
        elif isinstance(element, Sequence) and not isinstance(element, str):  # Since string are sequence we negate it to be log in the else
```
or 
`and not str` makes no sense.
```suggestion
        elif isinstance(element, Sequence) and not isinstance(element, AnyStr):  # Since string are sequence we negate it to be log in the else
```